### ISSUE TITLE
add tapo show "Invalid authentication data"

### ISSUE THREAD

**Issue by z59492085** [JurajNyiri/HomeAssistant-Tapo-Control#365]
### Description

add tapo show "Invalid authentication data"

### Reproduction Steps

add tapo show "Invalid authentication data"

### Expected behavior

can use add tapo c200

### Device Firmware

1.3.4

### Using stream component

Yes

### Does camera work via official integrations?

No

### Camera has all attributes filled out in developer tools

Yes

### HASS Environment

home assistant

### Search for similar issues

Yes

**Comment by JurajNyiri** [comment]
Enter correct authentication data.

**Comment by z59492085** [comment]
I tried this and it didn't change account
RTSP streaming works fine in VLC media player

**Comment by z59492085** [comment]
I have confirmed that there is no problem with the account... just updated a few days ago

**Comment by JurajNyiri** [comment]
Restart camera and make sure you are not hitting it with high load and try again. Make sure to read instructions carefully.

**Comment by flalmeida3105** [comment]
Same issue here, after removing it and trying to re-add it, it does not recognize the authentication. That occured just to 1 camera, all the other are working just fine. Is there anywhere I can clean cache or references to the old camera IP?

**Comment by JurajNyiri** [comment]
You can clean it in .storage, search for "tapo_control" and delete everything related to it, then reboot HA. That will not be the issue though. You need to reset the camera, not do anything with it and wait, it will start working.

**Comment by flalmeida3105** [comment]
@JurajNyiri 

First of all, thanks for helping out.

I tried your suggestion, and unfortunately it did not work as expected.

I kept troubleshooting, and theoretically, it has all the required information to connect to the camera, but, somehow, it does not recognize the authentication data/method/config/etc, not sure why. See logs below:

[ADD DEVICE][192.x.x.x] Showing config flow for Camera Account.
[ADD DEVICE][192.x.x.x] Verifying Camera Account.
[ADD DEVICE][192.x.x.x] Verifying ports all required camera ports.
[ADD DEVICE][192.x.x.x] All camera ports are opened.
[ADD DEVICE][192.x.x.x] Testing RTSP stream.
[isRtspStreamWorking][192.x.x.x] Testing RTSP stream.
[isRtspStreamWorking][192.x.x.x] Creating image frame.
[isRtspStreamWorking][192.x.x.x] Encoding username and password.
[isRtspStreamWorking][192.x.x.x] Getting image from rtsp://HIDDEN_USERNAME:HIDDEN_PASSWORD@192.x.x.x:554/stream1.
[custom_components.tapo_control] async_check_entities
[custom_components.tapo_control] [isRtspStreamWorking][192.x.x.x] Image data received.
[ADD DEVICE][192.x.x.x] RTSP stream returned invalid authentication data error.

In short, it works properly via TAPO App, via VLC and iSpy, it just does not work via HA. I've tested everything, and nothing worked, so Today, I decided to assign this camera a different IP, and to my surprise it worked.

I understand that caching should not be issue as you mentioned before, but these details (IP and credentials or something else) are kept in somewhere (HA configs, maybe?) that conflicts when a reset is required, maybe there is a proper way to remove the camera that cleans everything up, and I did not follow it.

Anyways, only after changing the camera's IP, I was able to set it up again on HA, not sure how relevant this info might be for you, but I thought it might be useful to others facing the same issue.

**Comment by JurajNyiri** [comment]
Hi, what probably happened is that when you changed IP whatever was creating the load previously resulting in stream not working was no longer creating the load as the IP was now not working.

**Comment by cynicer** [comment]
I'm experiencing the exact same issue, but on all my cameras at once. Didn't change any camera config, it just stopped working today (probably). Perhaps related to the last HA update, not sure. Video is fine, controls won't work with the same issue. Will try the troubleshooting steps, but assigning new IPs to all cameras doesn't sound like a solution..

**Comment by flalmeida3105** [comment]
@cynicer in my case it was okay because I was setting it up, but I 100% agree with you, this is a workaround and it cannot be considered a solution.

I'm in contact with tapo's support trying to figure out if this is an issue on their end.

**Comment by cynicer** [comment]
Would be surprising if this is an issue on their end. The stream works just fine and the integration has too, for months. Not really sure what changed, didn't have time to in depth troubleshoot it.

**Comment by cynicer** [comment]
In my case it was a layer 8 problem. I changed the cloud password a few months ago, but only now the token expired. Forgot to update it in home assistant.

**Comment by danktankk** [comment]
same problem all this time later.  this does not seem to have any rhyme or reason.  Now I cant get it to add the camera back at all.  why are so many people having such an issue with IP/user/pass?  this cant be on us for every post

**Comment by JurajNyiri** [comment]
The issue you are searching for:

    Is pinned under issues
    Has a section in main readme file with big title right at the top
    Has 3 red exclamations marks in the title
    Contains a workaround both for accounts still working and the ones affected.

And a warning was sent a month ago as an update via HACS.

You can find it at https://github.com/JurajNyiri/HomeAssistant-Tapo-Control/issues/551.

Please do not post me too comments or any other comment that does not provide any value to the discussion as everyone monitoring it gets notified. If the comments with no value continue maintainer will be forced to lock the issue to contributors only.
